variables.tf

```
@@ -144,6 +144,7 @@ variable "data_disks" {
     storage_account_type = optional(string, "Standard_LRS")
     caching              = optional(string, "ReadWrite")
     lun                  = optional(number)
+    hyper_v_generation   = optional(string, null)
```

`optional` sets the default value to `null` by default, so no need to set it explicitly:

```suggestion
    hyper_v_generation   = optional(string)
```

---

You are correct. No need for default null values.